File: SaluteOnline.Domain/Domain/EF/InnerMessage.cs
using System;
using SaluteOnline.Domain.Common;
using SaluteOnline.Domain.DTO;

namespace SaluteOnline.Domain.Domain.EF
{
    public class InnerMessage : IEntity
    {
        public Guid Guid { get; set; }
        public int Id { get; set; }

        public int? SenderId { get; set; }
        public EntityType SenderType { get; set; }

        public int? ReceiverId { get; set; }
        public EntityType ReceiverType { get; set; }

        public MessageStatus Status { get; set; }
        public string Title { get; set; }
        public string Body { get; set; }
        public DateTimeOffset Created { get; set; }
        public DateTimeOffset LastActivity { get; set; }

        public bool SentBySystem { get; set; }
        public bool OneResponseForAll { get; set; }
    }
}

File: SaluteOnline.Client/ClientApp/app/dto/innerMessage.ts
import { BaseFilter } from "./common";

export class InnerMessageDto {
    id: number;
    senderId: number | null;
    senderType: EntityType;
    receiverId: number | null;
    receiver: EntityType;
    status: MessageStatus;
    title: string;
    body: string;
    created: Date;
    lastActivity: Date;
    sentBySystem: boolean;
    oneResponseForAll: boolean;
    avatar: string;
    senderName: string;
}

export class InnerMessagesFilter extends BaseFilter {
    constructor(type: EntityType, id: number | null, status: MessageStatus) {
        super();
        this.receiverType = type;
        this.receiverId = id;
        this.status = status;
    }

    receiverId: number | null;
    receiverType: EntityType;
    status: MessageStatus;
}

export enum MessageStatus {
    None = 0,
    Pending = 1,
    Read = 2,
    Closed = 3,
    All = 4
}

export enum EntityType {
    System = 0,
    User = 1,
    Club = 2
}